python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -418,6 +418,94 @@ def test_actor_method_bind_same_input_attr(ray_start_regular):
     compiled_dag.teardown()
 
 
+def test_actor_method_bind_diff_input_attr_1(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # Two class methods are bound to two different input
+        # attribute nodes.
+        output1 = actor.inc.bind(inp[0])
+        output2 = actor.inc.bind(inp[1])
+        dag = MultiOutputNode([output1, output2])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(0, 1)
+    assert ray.get(ref) == [0, 1]
+
+    ref = compiled_dag.execute(1, 2)
+    assert ray.get(ref) == [2, 4]
+
+    compiled_dag.teardown()
+
+
+def test_actor_method_bind_diff_input_attr_2(ray_start_regular):
+    actor = Actor.remote(0)
+    with InputNode() as inp:
+        # Three class methods are bound to two different input
+        # attribute nodes. Two methods are bound to the same input
+        # attribute node.
+        output1 = actor.inc.bind(inp[0])
+        output2 = actor.inc.bind(inp[0])
+        output3 = actor.inc.bind(inp[1])
+        dag = MultiOutputNode([output1, output2, output3])
+    compiled_dag = dag.experimental_compile()
+    ref = compiled_dag.execute(0, 1)
+    assert ray.get(ref) == [0, 0, 1]
+
+    ref = compiled_dag.execute(1, 2)
```

Comment:
For all these tests, we should run multiple times to make sure the caching behavior is correct. Running once will likely miss catching issues.

I also recommend using `Collector.collect_two/three` when possible which is more powerful to check correctness.

Reply:
> For all these tests, we should run multiple times to make sure the caching behavior is correct. Running once will likely miss catching issues.

Do you mean calling `execute` more than once? Each test has already been executed twice.

> I also recommend using Collector.collect_two/three when possible which is more powerful to check correctness.

Good idea

Reply:
> Do you mean calling execute more than once? Each test has already been executed twice.

Yeah, maybe at least 3 times would be safer.

Reply:
update: https://github.com/ray-project/ray/pull/47489/commits/503351449a3d2462946fb2f899ba2d6eee789efd